content/4_Gold/BFS.mdx

```diff
@@ -576,6 +576,135 @@ int main() {
+
+            for (int i = 0; i < 4; i++) {
+                int nx = curr[0] + dx[i];
+                int ny = curr[1] + dy[i];
+
+                if(nx < 0 || ny < 0 || nx >= H || ny >= W) continue;
+                if(count[nx][ny] > 0) continue;
+                if(grid[nx][ny] == -1) continue;
+                if(grid[nx][ny] != grid[curr[0]][curr[1]]) {
+                    count[nx][ny] = count[curr[0]][curr[1]] + 1;
+                    q.addLast(new int[]{nx, ny});
+                }
+                else {
+                    count[nx][ny] = count[curr[0]][curr[1]];
+                    q.addFirst(new int[]{nx, ny});
+                }
+            }
+        }
+
+        return N;
+    }
+
+	//BeginCodeSnip{FastIO}
+    private static class FastIO extends PrintWriter {
+        private InputStream stream;
+        private byte[] buf = new byte[1<<16];
+        private int curChar, numChars;
+
+        // standard input
+        public FastIO() { this(System.in,System.out); }
+        public FastIO(InputStream i, OutputStream o) {
+            super(o);
+            stream = i;
+        }
+        // file input
+        public FastIO(String i, String o) throws IOException {
+            super(new FileWriter(o));
+            stream = new FileInputStream(i);
+        }
+
+        // throws InputMismatchException() if previously detected end of file
+        private int nextByte() {
+            if (numChars == -1) throw new InputMismatchException();
+            if (curChar >= numChars) {
+                curChar = 0;
+                try {
+                    numChars = stream.read(buf);
+                } catch (IOException e) {
+                    throw new InputMismatchException();
+                }
+                if (numChars == -1) return -1; // end of file
+            }
+            return buf[curChar++];
+        }
+
+        // to read in entire lines, replace c <= ' '
+        // with a function that checks whether c is a line break
+        public String next() {
+            int c; do { c = nextByte(); } while (c <= ' ');
+            StringBuilder res = new StringBuilder();
+            do { res.appendCodePoint(c); c = nextByte(); } while (c > ' ');
+            return res.toString();
+        }
+        public int nextInt() { // nextLong() would be implemented similarly
+            int c; do { c = nextByte(); } while (c <= ' ');
+            int sgn = 1; if (c == '-') { sgn = -1; c = nextByte(); }
+            int res = 0;
+            do {
+                if (c < '0' || c > '9')
+                    throw new InputMismatchException();
+                res = 10*res+c-'0';
+                c = nextByte();
+            } while (c > ' ');
+            return res * sgn;
+        }
+        public double nextDouble() { return Double.parseDouble(next()); }
+    }
+	//EndCodeSnip
+}
+```
+
+<Warning>
+
+Due to oj.uz's grading constraints for Java, this solution will TLE on the judge.
```

Comment:
Are there no constant optimizations you can do to prevent this TLE?

Reply:
I tried, but needs 100ms rn.

Reply:
also checked oj.uz, there were no Java ACs

Reply:
Maybe try CSES and if it passes there, just pretend oj.uz doesn't exist

Reply:
can u provide a link?

Reply:
@HarryWangATX https://cses.fi/104/list/

Reply:
TLEs as well on cses

Reply:
hmmmmmmmm. do u guys see any optimization spots?

Reply:
maybe try changing `int[]` to a custom `Pair` in the queue?

Reply:
still TLEs on the same testcases